controller/tests/test_plan_controller.py

```diff
@@ -0,0 +1,147 @@
+from controller.plan_controller import controller_error
+
+
+class PlanControllerTest(unittest.TestCase):
+    def setUp(self):
+        Plan.delete_all()
+
+    def test_plan_status(self):
+        """
+        Test status flag of plan. Should be false is plan is open.
+        """
+        plan = Plan(name='My Plan',
+                    emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                    description='Test emergency plan',
+                    geographical_area='',
+                    camps=[Camp(name='TestCamp')])
+        self.assertEqual(False, plan.close_plan_status)
+        self.setUp()
+
+    def test_close_plan_controller(self):
+        """
+        Test status flag of plan. Should be true when plan is closed.
+        """
+        plan = Plan(name='My Plan',
+                    emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                    description='Test emergency plan',
+                    geographical_area='',
+                    camps=[Camp(name='TestCamp')])
+        close_plan(plan)
+        self.assertEqual(True, plan.close_plan_status)
+        self.setUp()
+
+    def test_no_close_date(self):
+        """
+        Test that close date is not set during plan creation.
+        """
+        plan = Plan(name='My Plan',
+                    emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                    description='Test emergency plan',
+                    geographical_area='',
+                    camps=[Camp(name='TestCamp')])
+        self.assertEqual(None, plan.close_date)
+        self.setUp()
+
+    def test_close_date_set(self):
+        """
+        Test plan closure date is set to today's date when plan is closed.
+        """
+        self.setUp()
+        plan = Plan(name='My Plan',
+                    emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                    description='Test emergency plan',
+                    geographical_area='',
+                    camps=[Camp(name='TestCamp')])
+        close_plan(plan)
+        self.assertEqual(date.today(), plan.close_date)
+
+    def test_find_plan_controller(self):
+        """
+        Test to see find plan given plan name.
+        """
+        test_plan = Plan(name='My Plan',
+                         emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                         description='Test emergency plan',
+                         geographical_area='',
+                         camps=[Camp(name='TestCamp')])
+        found_plan = find_plan('My Plan')
+        self.assertEqual(test_plan, found_plan)
+        self.setUp()
+
+    def test_no_plan_found_controller(self):
+        """
+        Test that if no plan is found then an exception is raised.
+        """
+        with self.assertRaises(controller_error.ControllerError):
+            find_plan('My Plan')
+
+        self.setUp()
+
+    def test_not_plan_instance(self):
+        """
+        Test that if input is not a Plan type then exception is raised.
```

Comment:
I understood your point when I checked the next comment. I'll remove the test for this and also remove the exception handling done in find_plan as well since it's not necessary in that case.

Reply:
I agree based on the next comment, I'll remove this test!